zerver/tests/test_typing.py

```diff
@@ -0,0 +1,156 @@
+        """
+        Sending typing notification to yourself
+        is successful.
+        """
+        email = "hamlet@zulip.com"
+
+        events = [] # type: List[Dict[str, Any]]
+        with tornado_redirected_to_list(events):
+            result = self.client_post("/api/v1/typing", {"to": email},
+                                        **self.api_auth(email))
+        self.assert_json_success(result)
+        self.assertTrue(len(events) == 1)
+
+        event = events[0]['event']
+        display_recipient = get_display_recipient(event['recipient'])
+        if not isinstance(display_recipient, string_types):
+            event_recipient_emails = set(user['email'] for user in display_recipient)
+        else:
+            event_recipient_emails = set([display_recipient])
+
+        self.assertTrue(event['sender'] == get_user_profile_by_email(email))
+        self.assertTrue(event_recipient_emails == set([email]))
+        self.assertTrue(event['type'] == 'typing_notification')
+
+    def test_send_notification_to_another_user_event(self):
+        # type: () -> None
+        """
+        Sending typing notification to another user
+        is successful.
+        """
+        sender = "hamlet@zulip.com"
+        recipient = "othello@zulip.com"
+        events = [] # type: List[Dict[str, Any]]
+        with tornado_redirected_to_list(events):
+            result = self.client_post("/api/v1/typing", {"to": recipient},
+                                        **self.api_auth(sender))
+        self.assert_json_success(result)
+        self.assertTrue(len(events) == 1)
+
+        event = events[0]['event']
+        display_recipient = get_display_recipient(event['recipient'])
+        if not isinstance(display_recipient, string_types):
+            event_recipient_emails = set(user['email'] for user in display_recipient)
+        else:
+            event_recipient_emails = set([display_recipient])
+
+        self.assertTrue(event['sender'] == get_user_profile_by_email(sender))
+        self.assertTrue(event_recipient_emails == set([recipient]))
+        self.assertTrue(event['type'] == 'typing_notification')
+
+    def test_send_notification_without_reciever(self):
+        # type: () -> None
+        """
+        Sending typing notification without receiver fails
+        """
+        sender = "hamlet@zulip.com"
+        result = self.client_post("/api/v1/typing", {},
+                                        **self.api_auth(sender))
+        self.assert_json_error(result, "Notification reciever is required")
+
+    def test_send_notification_success_result(self):
+        # type: () -> None
+        """
+        Sending typing notification results in success
+        """
+        sender = "hamlet@zulip.com"
+        reciever = "othello@zulip.com"
+        result = self.client_post("/api/v1/typing", {"to": reciever},
+                                        **self.api_auth(sender))
+        self.assert_json_success(result)
```

Comment:
This test seems potentially unnecessary given the tests above, since it's just a subset of e.g. ` test_send_notification_to_another_user_event`, right?